sky/provision/kubernetes/utils.py

```diff
@@ -711,6 +711,13 @@ def check_tpu_fits(candidate_instance_type: 'KubernetesInstanceType',
             node.metadata.labels[gpu_label_key] == gpu_label_val
         ]
         assert gpu_nodes, 'GPU nodes not found'
+        # filter out nodes that do not have enough GPUs
+        gpu_nodes = [
+            node for node in gpu_nodes
+            if get_node_accelerator_count(node.status.allocatable) >= acc_count
+        ]
+        if len(gpu_nodes) == 0:
```

Comment:
nit: style guide: use implicit false to evaluate for empty sequences: https://google.github.io/styleguide/pyguide.html#2144-decision
```suggestion
        if not gpu_nodes:
```